services/horizon/internal/test/integration/integration.go

```diff
@@ -42,14 +44,41 @@ type Config struct {
 	ProtocolVersion       int32
 	SkipContainerCreation bool
 	CoreDockerImage       string
+
+	// Weird naming here because bools default to false, but we want to start
+	// Horizon by default.
+	DontStartHorizon bool
+
+	// If you want to override the default parameters passed to Horizon, you can
+	// set this map accordingly. All of them are passed along as --k=v, but if
+	// you pass an empty value, the parameter will be dropped. (Note that you
+	// should exclude the prepending `--` from keys; this is for compatibility
+	// with the constant names in flags.go)
+	//
+	// You can also control the environmental variables in a similar way, but
+	// note that CLI args take precedence over envvars, so set the corresponding
+	// CLI arg empty.
+	HorizonParameters  map[string]string
+	HorizonEnvironment map[string]string
+}
+
+type CaptiveConfig struct {
+	binaryPath string
+	configPath string
 }
 
 type Test struct {
-	t             *testing.T
+	t *testing.T
+
+	composePath string
+
 	config        Config
+	coreConfig    CaptiveConfig
 	horizonConfig horizon.Config
-	hclient       *sdk.Client
-	cclient       *stellarcore.Client
+
+	hclient *sdk.Client
```

Comment:
To match the above `horizonConfig`:
```suggestion
	horizonClient *sdk.Client
```